### PR TITLE
add unit tests for zookeeper orchestration center

### PR DESCRIPTION
**Pull request by menghaoranss**
For #4439 

Changes proposed in this pull request:
- add zookeeper unit tests

### CODE REVIEW

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperInstanceTest.java [hunk 1]
```diff
@@ -17,28 +17,44 @@
...
 import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 
 import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertThat;
 
 public class CuratorZookeeperInstanceTest {
```
Comment (terrymanu) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperInstanceTest.java]: Please add final if the class is not for extend

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperInstanceTest.java [hunk 2]
```diff
@@ -67,4 +83,114 @@ public void assertGetChildrenKeys() {
...
+        client.delete().forPath("/test/children/5");
+        DataChangedEvent dataChangedEvent = future.get(5, TimeUnit.SECONDS);
+        assertNotNull(dataChangedEvent);
+        assertThat(dataChangedEvent.getChangedType(), is(DataChangedEvent.ChangedType.DELETED));
+        assertThat(dataChangedEvent.getKey(), is("/test/children/5"));
+        assertThat(dataChangedEvent.getValue(), is("value5"));
+    }
+    
+    @Test
+    @SneakyThrows
```
Comment (terrymanu) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperInstanceTest.java]: Just throw exception from method signature, @SneakyThrows should use only in the situation for exception never occur

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperInstanceTest.java [hunk 3]
```diff
@@ -67,4 +83,114 @@ public void assertGetChildrenKeys() {
...
+    public void assertWatchAddedChangedType() {
+        curatorZookeeperInstance.persist("/test/children/4", "value4");
+        final AtomicReference<DataChangedEvent> actualDataChangedEvent = new AtomicReference<>();
+        curatorZookeeperInstance.watch("/test/children", dataChangedEvent -> actualDataChangedEvent.set(dataChangedEvent));
+        Thread.sleep(2000);
```
Comment (terrymanu) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperInstanceTest.java]: Please add `L` is the number is long type

Comment (menghaoranss) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperInstanceTest.java]: I will pay more attention next time.

**Pull request merged**